### PR TITLE
Add Synapse 101 and 201 quickstarts

### PR DESCRIPTION
**Pull request by murggu**
Includes:

- Synapse 101 quickstart
- Synapse 2021 quickstart

### CODE REVIEW

File: quickstart/101-synapse/.gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Terraform specific
```
Comment (grayzu) [inline quickstart/101-synapse/.gitignore]: Please remove these .gitignore files.

File: quickstart/101-synapse/locals.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+locals {
+  basename = "${var.name}-${var.environment}"
+  safe_basename = replace(local.basename, "-", "")
```
Comment (grayzu) [inline quickstart/101-synapse/locals.tf:3]: Love this approach!

File: quickstart/101-synapse/variables.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+variable "synadmin_username" {
+  type        = string
+  description = "Specifies The login name of the SQL administrator"
+  default     = "sqladminuser"
+}
+
+variable "synadmin_password" {
+  type        = string
+  description = "The Password associated with the sql_administrator_login for the SQL administrator"
+  default     = "ThisIsNotVerySecure!"
```
Comment (grayzu) [inline quickstart/101-synapse/variables.tf]: Please do not provide a default password here. A better approach would be to use the random provider to automatically generate a random password.

File: quickstart/201-synapse-secure/.gitignore [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Terraform specific
```
Comment (grayzu) [inline quickstart/201-synapse-secure/.gitignore]: Remove this file.

File: quickstart/101-synapse/main.tf [hunk 5]
```diff
@@ -0,0 +1,22 @@
+terraform {
```
Comment (grayzu) [inline quickstart/101-synapse/main.tf]: Please move the terraform and provider blocks into a providers.tf file.

File: quickstart/201-synapse-secure/main.tf [hunk 6]
```diff
@@ -0,0 +1,22 @@
+terraform {
```
Comment (grayzu) [inline quickstart/201-synapse-secure/main.tf]: Please move terraform and provider blocks to provider.tf file.

File: quickstart/201-synapse-secure/variables.tf [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+variable "jumphost_username" {
+  type        = string
+  description = "Admin username of the VM"
+  default     = "azureuser"
+}
+
+variable "jumphost_password" {
+  type        = string
+  description = "Password for the admin username of the VM"
+  default     = "ThisIsNotVerySecure!"
```
Comment (grayzu) [inline quickstart/201-synapse-secure/variables.tf]: See comment from 101 example.

File: quickstart/201-synapse-secure/variables.tf [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+variable "synadmin_username" {
+  type        = string
+  description = "Specifies The login name of the SQL administrator"
+  default     = "sqladminuser"
+}
+
+variable "synadmin_password" {
+  type        = string
+  description = "The Password associated with the sql_administrator_login for the SQL administrator"
+  default     = "ThisIsNotVerySecure!"
```
Comment (grayzu) [inline quickstart/201-synapse-secure/variables.tf]: See comment on default password from 101 example.

**Review by grayzu — CHANGES_REQUESTED** [review]
Thanks for these detailed quickstarts!!

Few minor changes requested but otherwise looks great!

File: quickstart/101-synapse/.gitignore (hunk 1 above)
Comment (murggu) [reply quickstart/101-synapse/.gitignore]: @grayzu done

File: quickstart/101-synapse/variables.tf (hunk 3 above)
Comment (murggu) [reply quickstart/101-synapse/variables.tf]: @grayzu done - moved to tfvars example

File: quickstart/201-synapse-secure/.gitignore (hunk 4 above)
Comment (murggu) [reply quickstart/201-synapse-secure/.gitignore]: @grayzu done

File: quickstart/101-synapse/main.tf (hunk 5 above)
Comment (murggu) [reply quickstart/101-synapse/main.tf]: @grayzu done

File: quickstart/201-synapse-secure/main.tf (hunk 6 above)
Comment (murggu) [reply quickstart/201-synapse-secure/main.tf]: @grayzu done

File: quickstart/201-synapse-secure/variables.tf (hunk 7 above)
Comment (murggu) [reply quickstart/201-synapse-secure/variables.tf]: @grayzu  done - moved to tfvars example

File: quickstart/201-synapse-secure/variables.tf (hunk 8 above)
Comment (murggu) [reply quickstart/201-synapse-secure/variables.tf]: @grayzu  done - moved to tfvars example

File: quickstart/101-synapse/README.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+| Name | Description | Default |
+|-|-|-|
+| name | Name of the deployment | - |
+| environment | The deployment environment name (used for pre- and postfixing resource names) | dev  |
+| location | The Azure region used for deployments | East US |
+| aad_admin.login | The login name of the Azure AD Administrator of the Synapse Workspace | - |
+| aad_admin.object_id| The object id of the Azure AD Administrator of the Synapse Workspace | - |
+| aad_admin.tenant_id| The tenant id of the Azure AD Administrator of the Synapse Workspace | - |
+| synadmin_username| Specifies the login name of the SQL administrator | sqladminuser |
+| synadmin_password| The Password associated with the synadmin_username for the SQL administrator | ThisIsNotVerySecure! |
```
Comment (grayzu) [inline quickstart/101-synapse/README.md]: After the changes, this no longer has a default so should be removed here as well.

File: quickstart/101-synapse/variables.tf [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+variable "aad_login" {
+  description = "AAD login"
+  type = object({
+    name      = string
+    object_id = string
+    tenant_id = string
+  })
+  default = {
```
Comment (grayzu) [inline quickstart/101-synapse/variables.tf]: Apologies for missing this on the first review but this should not have a default unless, you can provide values that will allow the configuration to succeed. Please remove the default.

File: quickstart/201-synapse-secure/README.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+| Name | Description | Default |
+|-|-|-|
+| name | Name of the deployment | - |
+| environment | The deployment environment name (used for pre- and postfixing resource names) | dev  |
+| location | The Azure region used for deployments | East US |
+| aad_admin.login | The login name of the Azure AD Administrator of the Synapse Workspace | - |
+| aad_admin.object_id| The object id of the Azure AD Administrator of the Synapse Workspace | - |
+| aad_admin.tenant_id| The tenant id of the Azure AD Administrator of the Synapse Workspace | - |
+| synadmin_username| Specifies the login name of the SQL administrator | sqladminuser |
+| synadmin_password| The Password associated with the synadmin_username for the SQL administrator | ThisIsNotVerySecure! |
```
Comment (grayzu) [inline quickstart/201-synapse-secure/README.md]: This is no longer the default so should be updated here as well.

File: quickstart/201-synapse-secure/README.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+| jumphost_username| Admin username of the VM | azureuser |
+| jumphost_password| Password for the admin username of the VM | ThisIsNotVerySecure! |
```
Comment (grayzu) [inline quickstart/201-synapse-secure/README.md]: This is no longer the default so should be updated here as well.

File: quickstart/201-synapse-secure/variables.tf [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+variable "aad_login" {
+  description = "AAD login"
+  type = object({
+    name      = string
+    object_id = string
+    tenant_id = string
+  })
+  default = {
```
Comment (grayzu) [inline quickstart/201-synapse-secure/variables.tf]: Apologies for missing this on the first review but this should not have a default unless, you can provide values that will allow the configuration to succeed. Please remove the default.

**Review by grayzu — CHANGES_REQUESTED** [review]
Couple more tweaks and will be good to merge.

Thanks!!

File: quickstart/201-synapse-secure/variables.tf (hunk 13 above)
Comment (murggu) [reply quickstart/201-synapse-secure/variables.tf]: @grayzu done

File: quickstart/201-synapse-secure/README.md (hunk 12 above)
Comment (murggu) [reply quickstart/201-synapse-secure/README.md]: @grayzu done

File: quickstart/201-synapse-secure/README.md (hunk 11 above)
Comment (murggu) [reply quickstart/201-synapse-secure/README.md]: @grayzu done

File: quickstart/101-synapse/variables.tf (hunk 10 above)
Comment (murggu) [reply quickstart/101-synapse/variables.tf]: @grayzu done

File: quickstart/101-synapse/README.md (hunk 9 above)
Comment (murggu) [reply quickstart/101-synapse/README.md]: @grayzu done

**Review by grayzu — APPROVED** [review]

**Pull request merged**